### ISSUE TITLE
Email sending fails with 'Invalid address "" '

### ISSUE THREAD

**Issue by crvio** [inventree/InvenTree#4769]
### Deployment Method

- [ ] Installer
- [ ] Docker Development
- [X] Docker Production
- [ ] Bare metal Development
- [ ] Bare metal Production
- [ ] Digital Ocean image
- [ ] Other (please provide a link `Steps to Reproduce`

### Describe the problem*

A fresh docker deployment of Inventree fails when trying to send an email to verify an email address.
When trying to resend the message for email verification from the admin profile settings page, I get a message of "Correo electrónico enviado a xxx@yyy.zz" (which roughly translates to "Email sent to xxx@yyy.zz", being that the email address it should be sent to.), but immediately a new error notification shows in the upper right corner.
When clicking on it, the following stacktrace gets shown: (pasted it in the relevant log output section)
By reading a bit into it I see that it is probably trying to email to an empty string address, yet the methods used belong to django modules, so tracing back the issue gets a little tricky for me.
Any help would be much appreciated.

### Steps to Reproduce

I followed the docs in [here](https://docs.inventree.org/en/latest/start/docker_prod/).
Logged in and started to set it up.
Email server, connection settings and credentials were input through the .env file pointed to from the docker-compose.yaml file.
When trying to add a new user (through the login page) no email was sent and the error was shown.
It might as well be the case that I'm missing some important step, but I couldn't find the error somewhere else with the solution as it usually is the case when I do.

### Relevant log output

```bash
Traceback (most recent call last):

File "/root/.local/lib/python3.9/site-packages/django/core/mail/message.py", line 82, in sanitize_address
token, rest = parser.get_mailbox(addr)

File "/usr/local/lib/python3.9/email/_header_value_parser.py", line 1791, in get_mailbox
token, value = get_name_addr(value)

File "/usr/local/lib/python3.9/email/_header_value_parser.py", line 1760, in get_name_addr
if value[0] in CFWS_LEADER:

IndexError: string index out of range


During handling of the above exception, another exception occurred:


Traceback (most recent call last):

File "/home/inventree/InvenTree/InvenTree/forms.py", line 277, in send_mail
result = super().send_mail(template_prefix, email, context)

File "/root/.local/lib/python3.9/site-packages/allauth/account/adapter.py", line 140, in send_mail
msg.send()

File "/root/.local/lib/python3.9/site-packages/django/core/mail/message.py", line 284, in send
return self.get_connection(fail_silently).send_messages([self])

File "/root/.local/lib/python3.9/site-packages/django/core/mail/backends/smtp.py", line 109, in send_messages
sent = self._send(message)

File "/root/.local/lib/python3.9/site-packages/django/core/mail/backends/smtp.py", line 121, in _send
from_email = sanitize_address(email_message.from_email, encoding)

File "/root/.local/lib/python3.9/site-packages/django/core/mail/message.py", line 84, in sanitize_address
raise ValueError('Invalid address "%s"' % addr)

ValueError: Invalid address ""
```

**Comment by crvio** [comment]
I've tried with 3 email addresses so far and always get the same error, one of them being the address of the admin account and the others of regular users.

**Comment by crvio** [comment]
sorry i closed it accidentally XD

**Comment by SchrodingersGat** [comment]
Have you had any emails successfully delivered by the server?

**Comment by crvio** [comment]
none so far

**Comment by SchrodingersGat** [comment]
@matmair have you ever seen this? First for me.

**Comment by matmair** [comment]
@crvio are you sure the server you are running on can sent mails? Can you verify that?
From the error message I would also check if you set the from_adress

**Comment by SchrodingersGat** [comment]
@crvio can you respond to the points above?

**Comment by crvio** [comment]
Yes!
Sorry for not answering promptly.
I have a docker deployment of nextcloud in the same server and mails can be sent from there.
I assumed that the setup could be similar, but it may be a false assumption.
Please let me know if there is additional setup for that.
The from_address field/config seems to be new for me.
I tried setting up the mail via the .env file that sits with the docker-compose.yaml file as follows:

# Outgoing email stuff
INVENTREE_EMAIL_HOST=mail.mydomain.com
INVENTREE_EMAIL_PORT=587
INVENTREE_EMAIL_USERNAME=inventree@mydomain.com
INVENTREE_EMAIL_PASSWORD=secret_pass
INVENTREE_EMAIL_SSL=False
INVENTREE_EMAIL_TLS=True

Probably i'm missing something.
Thanks in advance for you help.

**Comment by crvio** [comment]
Update:
Docker deployment of nextcloud is able to send emails with no encryption.
Same configuration in inventree fails with the following stacktrace:

```
Traceback (most recent call last):

File "/usr/local/lib/python3.9/smtplib.py", line 398, in getreply
line = self.file.readline(_MAXLINE + 1)

File "/usr/local/lib/python3.9/socket.py", line 704, in readinto
return self._sock.recv_into(b)

socket.timeout: timed out


During handling of the above exception, another exception occurred:


Traceback (most recent call last):

File "/home/inventree/InvenTree/InvenTree/forms.py", line 277, in send_mail
result = super().send_mail(template_prefix, email, context)

File "/root/.local/lib/python3.9/site-packages/allauth/account/adapter.py", line 140, in send_mail
msg.send()

File "/root/.local/lib/python3.9/site-packages/django/core/mail/message.py", line 284, in send
return self.get_connection(fail_silently).send_messages([self])

File "/root/.local/lib/python3.9/site-packages/django/core/mail/backends/smtp.py", line 102, in send_messages
new_conn_created = self.open()

File "/root/.local/lib/python3.9/site-packages/django/core/mail/backends/smtp.py", line 62, in open
self.connection = self.connection_class(self.host, self.port, **connection_params)

File "/usr/local/lib/python3.9/smtplib.py", line 255, in __init__
(code, msg) = self.connect(host, port)

File "/usr/local/lib/python3.9/smtplib.py", line 343, in connect
(code, msg) = self.getreply()

File "/usr/local/lib/python3.9/smtplib.py", line 401, in getreply
raise SMTPServerDisconnected("Connection unexpectedly closed: "

smtplib.SMTPServerDisconnected: Connection unexpectedly closed: timed out
```

**Comment by matmair** [comment]
@crvio do you have your own DNS server? It seems like the container can not reach the mail server or resolve its address.

**Comment by crvio** [comment]
So the last stacktrace was product of a misconfiguration of the SMTP port (strangely it works in NC; apparently it does use SSL, even though it's configured with no encryption, so i'll look later into that).
Once I corrected that, I was back to the 'Invalid address: "" ' problem, which was finally solved by configuring the INVENTREE_EMAIL_SENDER environment variable in the .env file, as you suggested initially.
So there it is. Now I'm able to send emails from the Inventree server.
Thanks a lot!
